src/core/a-assets.js

```diff
@@ -10,126 +10,127 @@ var warn = debug('core:a-assets:warn');
 /**
  * Asset management system. Handles blocking on asset loading.
  */
-module.exports = registerElement('a-assets', {
-  prototype: Object.create(ANode.prototype, {
-    createdCallback: {
-      value: function () {
-        this.isAssets = true;
-        this.fileLoader = fileLoader;
-        this.timeout = null;
-      }
-    },
-
-    attachedCallback: {
-      value: function () {
-        var self = this;
-        var i;
-        var loaded = [];
-        var mediaEl;
-        var mediaEls;
-        var imgEl;
-        var imgEls;
-        var timeout;
-
-        if (!this.parentNode.isScene) {
-          throw new Error('<a-assets> must be a child of a <a-scene>.');
-        }
-
-        // Wait for <img>s.
-        imgEls = this.querySelectorAll('img');
-        for (i = 0; i < imgEls.length; i++) {
-          imgEl = fixUpMediaElement(imgEls[i]);
-          loaded.push(new Promise(function (resolve, reject) {
-            // Set in cache because we won't be needing to call three.js loader if we have.
-            // a loaded media element.
-            THREE.Cache.files[imgEls[i].getAttribute('src')] = imgEl;
-            imgEl.onload = resolve;
-            imgEl.onerror = reject;
-          }));
-        }
+class AAssets extends ANode {
+  constructor (self) {
```

Comment:
The correct form here is:
```js
constructor(self) {
  self = super(self);
}
```

Both native and polyfilled browsers will receive the right upgraded context.

As it is now, you lose that upgrade in polyfilled browsers calling just `super(self)` without addressing its return.

This is explained in here https://github.com/WebReflection/document-register-element#v1-caveat and it's been working just fine since about ever in every other library based on DRE polyfill (StencilJS, Google AMP, etc)

Reply:
@justinfagnani That's a requirement of the polyfill as @WebReflection mentioned